enterprise/backend/src/metabase_enterprise/internal_user.clj

```diff
@@ -4,23 +4,34 @@
             [metabase.util.log :as log]
             [toucan2.core :as t2]))
 
-(defn- install-internal-user! []
-  (t2/insert-returning-instances!
-   User
-   {:id config/internal-mb-user-id
+(def ^:private internal-mb-user
+  {:id config/internal-mb-user-id
     :first_name "Metabase"
     :last_name "Internal"
     :email "internal@metabase.com"
     :password (str (random-uuid))
     :is_active false
     :is_superuser false
     :login_attributes nil
-    :sso_source nil}))
+    :sso_source nil})
+
+(defn- install-internal-user! []
+  (t2/insert-returning-instances! User internal-mb-user))
+
+(defn- internal-user-exists? []
+  (if-let [internal-user (t2/select-one User :email (:email internal-mb-user))]
+    (if (not= (:id internal-user) config/internal-mb-user-id)
+      (do
+        (log/info "Internal user already exists with an incorrect ID. Deleting...")
+        (t2/delete! User :email (:email internal-mb-user))
```

Comment:
Can't say I love having this delete! in `internal-user-exists?`. The value it returns is only used in an if branch, so why not put the do block on e.g. line 38? I think that would clear up the `if-let if` ... `false true false` bit too, it's kind of hard to follow.

Reply:
What do you think about doing the id-mismatch check when the internal user is about to be created?

Reply:
I'm not sure we need to delete the internal user. If it exists, just don't try to create it again?

Reply:
Yeah I'm going to revert this change